sentry-spring/src/main/java/io/sentry/spring/webflux/SentryWebFilter.java

```diff
@@ -19,27 +21,30 @@
 /** Manages {@link io.sentry.Scope} in Webflux request processing. */
 @ApiStatus.Experimental
 public final class SentryWebFilter implements WebFilter {
-  private final @NotNull IHub hub;
+
   private final @NotNull SentryRequestResolver sentryRequestResolver;
 
   public SentryWebFilter(final @NotNull IHub hub) {
-    this.hub = Objects.requireNonNull(hub, "hub is required");
+    Objects.requireNonNull(hub, "hub is required");
     this.sentryRequestResolver = new SentryRequestResolver(hub);
   }
 
   @Override
   public Mono<Void> filter(
       final @NotNull ServerWebExchange serverWebExchange,
       final @NotNull WebFilterChain webFilterChain) {
+    @NotNull IHub requestHub = Sentry.cloneMainHub();
     return webFilterChain
         .filter(serverWebExchange)
         .doFinally(
             __ -> {
-              hub.popScope();
+              requestHub.popScope();
+              Sentry.setCurrentHub(NoOpHub.getInstance());
             })
         .doFirst(
             () -> {
-              hub.pushScope();
+              Sentry.setCurrentHub(requestHub);
```

Comment:
Need to set it as current hub for the thread as `SentryScheduleHook` clones the current hub and without this it'd clone a random hub breaking scope.